src/main/java/com/redhat/devtools/lsp4ij/settings/ui/LanguageServerPanel.java

```diff
@@ -296,6 +310,21 @@ public JsonTextField getConfiguration() {
         return configurationWidget;
     }
 
+    public String getConfigurationSchemaContent() {
+        return configurationSchemaContent;
+    }
+
+    public void setConfigurationSchemaContent(String configurationSchemaContent) {
+        this.configurationSchemaContent = configurationSchemaContent;
+        if (StringUtils.isNotBlank(configurationSchemaContent)) {
+            getConfiguration().associateWithJsonSchema(configurationSchemaContent);

```

Comment:
Does anything validate the JSON schema itself?

Reply:
Using in the editor

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",

```

provides JSON Schema support:

![image](https://github.com/user-attachments/assets/c9f3369a-9e59-447b-a9e2-88433989fb71)

It works out of the box.

Reply:
Following up on this. In the spirit of "who will watch the watchers?", it seems like there needs to be some minimal validation of the JSON schema or there will be harder-to-diagnose downstream issues when we try to use an invalid JSON schema to validate an actual JSON document.